### PR TITLE
Core/Auras: implemented PROC_FLAG_HEARTBEAT and moved food/drink emote mechanic into heartbeat handling

### PR DESCRIPTION
**Pull request by Ovahlord**
**Changes proposed:**

-  PROC_FLAG_HEARTBEAT does now fire on every heartbeat
-  moved periodic food and drink emote handling into unitAura and use spellspecifctype to determine its type

**Tests performed:**
- tested ingame

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 1]
```diff
@@ -2614,6 +2614,28 @@ void UnitAura::AddStaticApplication(Unit* target, uint32 effMask)
...
+}
+
+void UnitAura::HandlePeriodicFoodSpellVisualKit()
+{
+    SpellSpecificType specificType = GetSpellInfo()->GetSpellSpecific();
+
+    bool food = specificType == SPELL_SPECIFIC_FOOD || specificType == SPELL_SPECIFIC_FOOD_AND_DRINK;
+    bool drink = specificType == SPELL_SPECIFIC_DRINK || specificType == SPELL_SPECIFIC_FOOD_AND_DRINK;
+
+    if (food)
```
Comment (moJIto-ice) [inline src/server/game/Spells/Auras/SpellAuras.cpp:2632]: use switch by specificType  ?

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -501,6 +501,19 @@ void Unit::Heartbeat()
 
     // SMSG_FLIGHT_SPLINE_SYNC for cyclic splines
     SendFlightSplineSyncUpdate();
+
+    // Aura Heartbeat
+    TriggerAuraHeartbeat();
```
Comment (moJIto-ice) [inline src/server/game/Entities/Unit/Unit.cpp:506]: is its rly usefull comment?

File: src/server/game/Entities/Unit/Unit.cpp [hunk 3]
```diff
@@ -501,6 +501,19 @@ void Unit::Heartbeat()
...
+}
+
+void Unit::TriggerAuraHeartbeat()
+{
+    // There are some auras which do invoke generic behavior on heartbeat
+    for (auto const& [_, auraApplication] : m_appliedAuras)
+        auraApplication->GetBase()->Heartbeat();
+
+    // PROC_FLAG_HEARTBEAT
```
Comment (moJIto-ice) [inline src/server/game/Entities/Unit/Unit.cpp:515]: is its rly usefull comment?

Comment (Ovahlord) [reply src/server/game/Entities/Unit/Unit.cpp:515]: it's called consistency and making sure that if someone ever starts messing around with it, that something remains to tell what all this was suposed to mean.

Comment (moJIto-ice) [reply src/server/game/Entities/Unit/Unit.cpp:515]: is possible to understand it from args .... Unit::ProcSkillsAndAuras(.....PROC_FLAG_HEARTBEAT

File: src/server/game/Spells/Auras/SpellAuras.cpp (hunk 1 above)
Comment (Ovahlord) [reply src/server/game/Spells/Auras/SpellAuras.cpp:2632]: I prefer clean readable code above a fallthrough case mess

**Pull request merged**